File: src/main/java/org/kowboy/util/NumberUtils.java
package org.kowboy.util;

/**
 * Utility functions for strings representing numbers. Because Java hates data.
 *
 * @author Craig McDaniel
 * @since 1.0
 */
public final class NumberUtils {
    public static final boolean isInteger(String s) {
        if (s == null) return false;

        try {
            int i = Integer.parseInt(s);
        } catch (NumberFormatException e) {
            return false;
        }
        return true;
    }

    /**
     * Inclusive range check for a given String argument.
     *
     * @param arg The string to parse.
     * @param min The minimum of the range check (inclusive).
     * @param max The maximum of the range check (inclusive).
     * @return <code>true</code> if the argument is an integer string and is between <code>min</code> and
     *          <code>max</code>, otherwise <code>false</code>.
     */
    public static boolean isBetween(String arg, int min, int max) {
        if (arg == null) return false;

        try {
            int i = Integer.parseInt(arg);
            return min <= i && i <= max;
        } catch (NumberFormatException e) {}

        return false;
    }
}

File: src/main/kotlin/org/kowboy/util/NumberUtils.kt
package org.kowboy.util

/**
 * Utility functions for strings representing numbers. Because Java hates data.
 *
 * @author Craig McDaniel
 * @since 1.0
 */
object NumberUtils {
    fun isInteger(s: String?): Boolean {
        if (s == null) return false
        try {
            s.toInt()
        } catch (e: NumberFormatException) {
            return false
        }
        return true
    }

    /**
     * Inclusive range check for a given String argument.
     *
     * @param arg The string to parse.
     * @param min The minimum of the range check (inclusive).
     * @param max The maximum of the range check (inclusive).
     * @return `true` if the argument is an integer string and is between `min` and
     * `max`, otherwise `false`.
     */
    fun isBetween(arg: String?, min: Int, max: Int): Boolean {
        if (arg == null) return false
        try {
            val i = arg.toInt()
            return i in min..max
        } catch (e: NumberFormatException) {
        }
        return false
    }
}